modules/cudaarithm/src/cuda/absdiff_scalar.cu

```diff
@@ -56,14 +56,14 @@ void absDiffScalar(const GpuMat& src, cv::Scalar val, bool, GpuMat& dst, const G
 
 namespace
 {
-    template <typename T, typename S> struct AbsDiffScalarOp : unary_function<T, T>
+    template <typename SrcType, typename ScalarType> struct AbsDiffScalarOp : unary_function<SrcType, SrcType>
```

Comment:
`T` is not "src" type only. There is "dst" type too. So "T" naming seems valid.
`S` is probably "work type" here ("WT").